### PR TITLE
More runtime data

### PR DESCRIPTION
**Pull request by bergie**
Follow-up to #695: save all components and graphs from runtime that match the project's namespace

### CODE REVIEW

File: components/RuntimeToProject.coffee [hunk 1]
```diff
@@ -1,6 +1,34 @@
 noflo = require 'noflo'
 uuid = require 'uuid'
 
+fetchSources = (components, runtime, sources, callback) ->
+  return callback null, sources unless components.length
+  handleMessage = (msg) ->
+    if msg.command is 'error'
+      callback msg.payload
```
Comment (jonnor) [inline components/RuntimeToProject.coffee]: Should turn `msg.payload` into an `Error` object?

File: components/RuntimeToProject.coffee [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+      return
+    if msg.command is 'source'
+      sources.push msg.payload
+      fetchSources components, runtime, sources, callback
```
Comment (jonnor) [inline components/RuntimeToProject.coffee]: Really don't like recursive async code, but I guess it works...

File: components/RuntimeToProject.coffee (hunk 1 above)
Comment (bergie) [inline components/RuntimeToProject.coffee]: Done in 08de71f

File: components/RuntimeToProject.coffee [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+  component = components.shift()
+  runtime.sendComponent 'getsource',
+    name: component
+
+fetchFromLibrary = (runtime, callback) ->
+  return callback null, [] unless runtime.isConnected()
+  return callback null, [] unless runtime.canDo 'component:getsource'
+  return callback null, [] unless runtime.definition?.namespace
+  return callback null, [] unless runtime.definition.components
+  components = Object.keys(runtime.definition.components).filter (componentName) ->
```
Comment (jonnor) [inline components/RuntimeToProject.coffee]: This code would be clearer if the filter function was moved out and given a name. `componentBelongsToProject` or similar

Comment (bergie) [inline components/RuntimeToProject.coffee]: Done

**Pull request merged**